# 1 ejercicio

Comenzamos importando las bibliotecas necesarias e inicializando el texto

In [21]:
import nltk
import re
from nltk.probability import FreqDist


file_path = 'NLP.txt'
file = open(file_path, 'r')

text = file.read()

In [12]:
text

"We can't say that we often think about the subtleties of our language. It is an intuitive behavior that relies on semantic cues such as words, signs, or images to convey information and meaning. It is sometimes said that learning a language is easier in adolescence because it is a repeatable behavior that can be instilled, much like walking. Also, languages do not follow strict rules and have many exceptions.\n\nLanguage acquisition, while natural for humans, is extremely complex for computers because it has a lot of unstructured data, a lack of strict rules, and no concrete perspective or context. Yet, humans increasingly rely on computer systems to communicate and perform tasks. This is why machine learning (ML) and artificial intelligence (AI) are gaining so much attention and popularity. Natural language processing is following the development of AI. Although these two terms may conjure up images of futuristic robots, there are already simple applications of natural language proce

In [22]:
# Remueve los signos de puntuación
text = re.sub(r'[^\w\s]', '', text)

# Tokenizamos el texto
words = nltk.word_tokenize(text)
words

['We',
 'cant',
 'say',
 'that',
 'we',
 'often',
 'think',
 'about',
 'the',
 'subtleties',
 'of',
 'our',
 'language',
 'It',
 'is',
 'an',
 'intuitive',
 'behavior',
 'that',
 'relies',
 'on',
 'semantic',
 'cues',
 'such',
 'as',
 'words',
 'signs',
 'or',
 'images',
 'to',
 'convey',
 'information',
 'and',
 'meaning',
 'It',
 'is',
 'sometimes',
 'said',
 'that',
 'learning',
 'a',
 'language',
 'is',
 'easier',
 'in',
 'adolescence',
 'because',
 'it',
 'is',
 'a',
 'repeatable',
 'behavior',
 'that',
 'can',
 'be',
 'instilled',
 'much',
 'like',
 'walking',
 'Also',
 'languages',
 'do',
 'not',
 'follow',
 'strict',
 'rules',
 'and',
 'have',
 'many',
 'exceptions',
 'Language',
 'acquisition',
 'while',
 'natural',
 'for',
 'humans',
 'is',
 'extremely',
 'complex',
 'for',
 'computers',
 'because',
 'it',
 'has',
 'a',
 'lot',
 'of',
 'unstructured',
 'data',
 'a',
 'lack',
 'of',
 'strict',
 'rules',
 'and',
 'no',
 'concrete',
 'perspective',
 'or',
 'context',
 'Yet',
 'h

Calcula la frecuencia de las palabras

In [23]:
freq_dist = FreqDist(words)
print(freq_dist)


<FreqDist with 159 samples and 250 outcomes>


## 2 ejercicio


Importa las stopwords y el algoritmo de stemming que deseas utilizar (en este caso, PorterStemmer)

In [30]:
from nltk.corpus import stopwords
from nltk.stem import PorterStemmer


stop_words = set(stopwords.words('english'))
ps = PorterStemmer()
ps

<PorterStemmer>

In [31]:
filtered_words = [ps.stem(w) for w in words if not w in stop_words]
filtered_words

['we',
 'cant',
 'say',
 'often',
 'think',
 'subtleti',
 'languag',
 'it',
 'intuit',
 'behavior',
 'reli',
 'semant',
 'cue',
 'word',
 'sign',
 'imag',
 'convey',
 'inform',
 'mean',
 'it',
 'sometim',
 'said',
 'learn',
 'languag',
 'easier',
 'adolesc',
 'repeat',
 'behavior',
 'instil',
 'much',
 'like',
 'walk',
 'also',
 'languag',
 'follow',
 'strict',
 'rule',
 'mani',
 'except',
 'languag',
 'acquisit',
 'natur',
 'human',
 'extrem',
 'complex',
 'comput',
 'lot',
 'unstructur',
 'data',
 'lack',
 'strict',
 'rule',
 'concret',
 'perspect',
 'context',
 'yet',
 'human',
 'increasingli',
 'reli',
 'comput',
 'system',
 'commun',
 'perform',
 'task',
 'thi',
 'machin',
 'learn',
 'ml',
 'artifici',
 'intellig',
 'ai',
 'gain',
 'much',
 'attent',
 'popular',
 'natur',
 'languag',
 'process',
 'follow',
 'develop',
 'ai',
 'although',
 'two',
 'term',
 'may',
 'conjur',
 'imag',
 'futurist',
 'robot',
 'alreadi',
 'simpl',
 'applic',
 'natur',
 'languag',
 'process',
 'everyday

# 3 ejercicio


Calcula el sentimiento de tu texto

In [28]:
from textblob import TextBlob
blob = TextBlob(text)
print(blob.sentiment)


Sentiment(polarity=0.028859126984126987, subjectivity=0.42723214285714284)


The output you provided, Sentiment(polarity=0.028859126984126987, subjectivity=0.42723214285714284), seems to represent sentiment analysis results for a given text. Sentiment analysis is the process of determining the sentiment or emotional tone of a text, usually categorized as positive, negative, or neutral.

In the output you provided, there are two values:

polarity: It represents the sentiment polarity of the text on a scale from -1 to 1. A value closer to 1 indicates positive sentiment, closer to -1 indicates negative sentiment, and closer to 0 indicates neutral sentiment. In this case, the polarity is approximately 0.0289, indicating a slightly positive sentiment.
subjectivity: It represents the subjectivity of the text on a scale from 0 to 1. A value closer to 0 indicates objective or factual text, and a value closer to 1 indicates subjective or opinionated text. In this case, the subjectivity is approximately 0.4272, indicating a moderately subjective text.
Please note that without the context of the actual text being analyzed, it's difficult to provide more specific interpretation or understanding of the sentiment analysis results.